hrm-domain/lambdas/custom-integrations/uscr-dispatcher/index.ts

```diff
@@ -14,17 +14,115 @@
+    const envResult = validateEnvironment();
+    if (isErr(envResult)) {
+      const message = `${envResult.error} ${envResult.message}`;
+      logger({ message, severity: 'error' });
+      return { statusCode: 500, body: message };
+    }
+
+    const payloadResult = validatePayload(JSON.parse(event.body || '{}'));
+    if (isErr(payloadResult)) {
+      const message = `${payloadResult.error} ${payloadResult.message}`;
+      logger({ message, severity: 'warn' });
+      return { statusCode: 400, body: message };
+    }
+
+    const headersResult = validateHeaders(event.headers);
+    if (isErr(headersResult)) {
+      const message = `${headersResult.error} ${headersResult.message}`;
+      logger({ message, severity: 'warn' });
+      return { statusCode: 400 };
+    }
+
+    const authResult = await authenticateRequest({
+      accountSid: payloadResult.data.accountSid,
+      authHeader: headersResult.data.authToken,
+      environment: envResult.data.environment,
+    });
+    if (isErr(authResult)) {
+      const message = authResult.error + authResult.message;
+      logger({ message, severity: 'warn' });
+      return { statusCode: 401 };
+    }
+
+    // Get (or create) case associated with the contact so we can track the incident being reported
+    const createCaseResult = await hrmService.getOrCreateCase({
+      accountSid: payloadResult.data.accountSid,
+      casePayload: payloadResult.data.casePayload,
+      contactId: payloadResult.data.contactId,
+      token: authResult.data.token,
+    });
+    if (isErr(createCaseResult)) {
+      const message = createCaseResult.error + createCaseResult.message;
+      logger({ message, severity: 'error' });
+      return { statusCode: 500 };
+    }
+
+    const { contact, caseObj } = createCaseResult.data;
+
+    logger({ message: JSON.stringify(createCaseResult), severity: 'info' });
+
+    // Case already contains a corresponding case entry section, we asume the incident has been created but something went wrong updating HRM. Poller will eventually bring consitency to this case
+    if (hrmService.hasIncidentCaseSection(caseObj)) {
+      logger({ message: 'case already has associated incident', severity: 'info' });
+      return {
+        statusCode: 200,
+      };
+    }
+
+    // Case does not contains a corresponding case entry section, we assume the incident was never reported (this can only happen if Beacon responded with an error)
+    const createIncidentResult = await beaconService.createIncident({
+      environment: envResult.data.environment,
+      incidentParams: mapping.toCreateIncident({
+        caseObj: caseObj,
+        contact,
+      }),
+    });
+    if (isErr(createIncidentResult)) {
+      const message = createIncidentResult.error + createIncidentResult.message;
+      logger({ message, severity: 'error' });
+      return { statusCode: 500 };
+    }
+
+    logger({ message: JSON.stringify(createIncidentResult), severity: 'info' });
```

Comment:
We don't want to be logging out full JSON objects in 'info' level, this should be debug